sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/AbstractAggregator.java

```diff
@@ -16,14 +16,22 @@
 
 package io.opentelemetry.sdk.metrics.aggregator;
 
+import java.util.concurrent.atomic.AtomicBoolean;
+
 abstract class AbstractAggregator implements Aggregator {
+  // Note: This is not 100% thread-safe. There is a race condition where recordings can
+  // be made in the moment between the reset and the setting of this field's value. In those
+  // cases, it is possible that a recording could be missed in a given recording interval, but
+  // it should be picked up in the next, assuming that more recordings are being made.
+  private final AtomicBoolean hasRecordings = new AtomicBoolean(false);
```

Comment:
Since we're not using CAS, this can just be `volatile boolean`.

Reply:
true enough. I tend to usually not use volatile, rather than the Atomics, but I don't have a really strong feeling either way. I think the performance is pretty much equal, IIRC. Although I have this feeling that that changed with java 14 and Atomics got much worse (/me goes to read his Java Specialists newsletters).

Reply:
updated.